### PR TITLE
Adds styles to SectionHeadline - issue57

### LINKED ISSUE #57: Add SectionHeadline
# Task

### Visuals
Screenshot_2020-05-04 Afternoon – Habitify Web

The content will be passed via prop. The color should be whatever the color of the background is, and just make it a bit transparent. The task should only focus on the text, don't worry about the background.

### notes: 
- font-weight: 500;
- text-transform: uppercase;
- font-size: 13px;
- color: hsla(0,0%,100%,.3);
- letter-spacing: 0;
- padding: 11px 15px;

#### Challenges:
- styling
- props (the content of button, in this example "MORE" will be passed via the children prop)

## Getting started
Since this is an isolated component you will develop, you will develop it in storybook.

1. Go to master branch, and pull the latest code
2. Make a new branch `git branch -b [branchname]`
3. Run `npm install`
4. Run `npm run storybook` 
5. Go to http://localhost:9009/?path=/story/sectionheadline--section-headline-story
6. Make some changes in [src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx](https://github.com/HandsOnLearning/the-project/blob/master/src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx)
7. See how it changes in storybook
8. Start coding :)

### PR DESCRIPTION
**Pull request by limbovski**
Added styles to SectionHeadline
Added an index to SectionHeadline story
Added GreySeeThru to Color variables

closes #57

### CODE REVIEW

File: src/components/Sidebar/components/SectionHeadline/SectionHeadline.stories.tsx [hunk 1]
```diff
@@ -17,8 +17,9 @@ const colors = [
 export const SectionHeadlineStory = () => (
   <div>
     This is the component on four different backgrounds.
-    {colors.map((color) => (
+    {colors.map((color, index) => (
       <div
+        key={index}
```
Comment (MoSattler) [inline src/components/Sidebar/components/SectionHeadline/SectionHeadline.stories.tsx:22]: haha, well done, you're fixing my code.

File: src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
+import { Color } from "variables";
 
 type Props = {
   children: ReactNode;
 };
 
+// Getting an error when trying to use textTransform property
+const sectionHeadStyles = {
+  fontWeight: 500,
+  // textTransform: '',
```
Comment (MoSattler) [inline src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx:11]: whats the error?

File: src/components/Sidebar/components/SectionHeadline/SectionHeadline.stories.tsx (hunk 1 above)
Comment (limbovski) [reply src/components/Sidebar/components/SectionHeadline/SectionHeadline.stories.tsx:22]: Ah lots to learn... just working my way through it all.

File: src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx (hunk 2 above)
Comment (limbovski) [reply src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx:11]: On style:

Type '{ fontWeight: number; textTransform: string; fontSize: string; color: Color; letterSpacing: number; padding: string; }' is not assignable to type 'CSSProperties'.
  Types of property 'textTransform' are incompatible.
    Type 'string' is not assignable to type '"uppercase" | "-moz-initial" | "inherit" | "initial" | "revert" | "unset" | "none" | "capitalize" | "full-size-kana" | "full-width" | "lowercase" | undefined'.ts(2322)
index.d.ts(1763, 9): The expected type comes from property 'style' which is declared here on type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>'

Screenshot 2020-05-05 at 2 55 40

Comment (MoSattler) [reply src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx:11]: weird! I take a look

Comment (MoSattler) [reply src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx:11]: So the issue is described [here](https://github.com/microsoft/TypeScript/issues/21618#issuecomment-362945112).
It's a bit complicated, but it's a TypeScript issue, and you shouldnt worry abot it too much right now. I will suggest a solution in a min.

File: src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
+  fontSize: "13px",
+  color: Color.GreySeeThru,
+  letterSpacing: 0,
+  padding: "11px 15px",
+};
+
```
Comment (MoSattler) [inline src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx:17]:
```suggestion
```

File: src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx [hunk 4]
```diff
@@ -1,20 +1,22 @@
...
-  return (
-    <div
-      css={`
-        // Give me some styles!
-      `}
-    >
-      passed via prop: {children}
-    </div>
-  );
+  return <div style={sectionHeadStyles}>{children}</div>;
```
Comment (MoSattler) [inline src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx:19]:
```suggestion
  return <div style={{
  fontWeight: 500,
  // textTransform: 'uppercase',
  fontSize: "13px",
  color: Color.GreySeeThru,
  letterSpacing: 0,
  padding: "11px 15px",
}}>{children}</div>;
```

**Comment by MoSattler** [comment]
Once you fixed the the text transform, can you take a swing at [this](https://github.com/HandsOnLearning/the-project/pull/58#pullrequestreview-405392074)?

File: src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx (hunk 2 above)
Comment (limbovski) [reply src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx:11]: Thanks. Yes I knew it was something odd. But I didn't really know where to look to fix it. This is my first time using react obviously but also I have zero experience with TypeScript. Thanks.

File: src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx (hunk 3 above)
Comment (limbovski) [reply src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx:17]: Thanks Mo. What's the best way to implement your change here?

File: src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx (hunk 4 above)
Comment (limbovski) [reply src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx:19]: Thanks Mo. What's the best way to implement your change here?

**Comment by limbovski** [comment]
> Once you fixed the the text transform, can you take a swing at [this](https://github.com/HandsOnLearning/the-project/pull/58#pullrequestreview-405392074)?

Will do.

File: src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx (hunk 4 above)
Comment (MoSattler) [reply src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx:19]: you can either click "Commit suggestion" above, or copy paste the code in your editor.

Comment (MoSattler) [reply src/components/Sidebar/components/SectionHeadline/SectionHeadline.tsx]: I just did that for you really quickly

**Comment by MoSattler** [comment]
@all-contributors please add @limbovski  for Code

**Pull request merged**